TOMM20.py

```diff
@@ -0,0 +1,263 @@
+    'CUC': 'L', 'CUA': 'L', 'CUG': 'L', 'AUU': 'I', 'AUC': 'I',
+    'AUA': 'I', 'GUU': 'V', 'GUC': 'V', 'GUA': 'V', 'GUG': 'V',
+    'UCU': 'S', 'UCC': 'S', 'UCA': 'S', 'UCG': 'S', 'AGU': 'S',
+    'AGC': 'S', 'CCU': 'P', 'CCC': 'P', 'CCA': 'P', 'CCG': 'P',
+    'ACU': 'T', 'ACC': 'T', 'ACA': 'T', 'ACG': 'T', 'GCU': 'A',
+    'GCC': 'A', 'GCA': 'A', 'GCG': 'A', 'UAU': 'Y', 'UAC': 'Y',
+    'UAA': 'stop', 'UAG': 'stop', 'UGA': 'stop', 'CAU': 'H',
+    'CAC': 'H', 'CAA': 'Q', 'CAG': 'Q', 'AAU': 'N', 'AAC': 'N',
+    'AAA': 'K', 'AAG': 'K', 'GAU': 'D', 'GAC': 'D', 'UGG': 'W',
+    'GAA': 'E', 'GAG': 'E', 'UGU': 'C', 'UGC': 'C', 'CGU': 'R',
+    'CGC': 'R', 'CGA': 'R', 'CGG': 'R', 'AGA': 'R', 'AGG': 'R',
+    'GGU': 'G', 'GGC': 'G', 'GGA': 'G', 'GGG': 'G', 'AUG': 'M'
+}
+
+DICT_MOLECULAR_MASS = {
+    'G': 75, 'A': 89, 'V': 117, 'L': 131, 'I': 131, 'P': 115,
+    'F': 165, 'Y': 181, 'W': 204, 'S': 105, 'T': 119, 'C': 121,
+    'M': 149, 'N': 132, 'Q': 146, 'D': 133, 'E': 147, 'K': 146,
+    'R': 174, 'H': 155
+}
+
+
+class BiologicalSequence(str):
+    """
+    Abstract class for working with biological sequences
+    Attributes of BiologicalSequence class:
+    -   seq (str): biological sequence
+    -   seq_type (str): which sequence DNA, RNA or protein the sequence is
+    """
+
+    def __init__(self, seq):
+        self.seq = seq
+        self.seq_type = None
+
+    def __len__(self):
+        return len(self.seq)
+
+    def __getitem__(self, idx):
+        return self.seq[idx]
+
+    def __getslice__(self, start, end):
+        return self.seq[start: end]
+
+    def __repr__(self):
+        return f'The sequence is: {self.seq}, type is {self.seq_type}'
+
+    def __str__(self):
+        return self.seq
+
+    def check_seq_type(self, check_type):
+        """
+        Check if sequence is check_type of molecule DNA, RNA or protein
+        Input:
+        -   check_type (str): type of molecule DNA, RNA, protein
+        Return:
+        -   bool: True is type match with check_type, False if not
+        """
+
+        seq_types = {
+            'DNA': ALPHABET_FOR_DNA,
+            'RNA': ALPHABET_FOR_RNA,
+            'Protein': ALPHABET_FOR_PROTEIN
+        }
+
+        if check_type not in seq_types:
+            raise ValueError(f'There is not such sequence type as {check_type}!')
+        return set(self.seq) <= seq_types[check_type]
+
+    def type_definition(self):
+        """
+        Set sequence type (DNA, RNA, protein).
+        Input:
+        -   seq (str): sequence that we need to get type:
+        Return:
+        -   type (str): type that sequence are
+        """
+        if self.check_seq_type('DNA'):
+            self.seq_type = 'DNA'
+        elif self.check_seq_type('RNA'):
+            self.seq_type = 'RNA'
+        elif self.check_seq_type('Protein'):
+            self.seq_type = 'Protein'
+        else:
+            raise ValueError(f'Sequence {self.seq} can not be analysed!')
```

Comment:
Можно еще вот так, и когда сравнение больше чем из 2х случаев, работает чуть быстрее
```suggestion
match True:
    case self.check_seq_type('DNA'):
        self.seq_type = 'DNA'
    case self.check_seq_type('RNA'):
        self.seq_type = 'RNA'
    case self.check_seq_type('Protein'):
        self.seq_type = 'Protein'
    case _:
        raise ValueError(f'Sequence {self.seq} can not be analysed!')
```